components/builder-worker/src/server.rs

```diff
@@ -177,17 +191,21 @@ impl Server {
     }
 
     fn set_busy(&mut self) -> Result<()> {
-        self.hb_cli.set_busy()?;
         self.state = State::Busy;
```

Comment:
I'd just move `self.state = State::Busy` to where we are calling `set_busy` and remove the function.